model/src/payouts/quanto.rs

```diff
@@ -0,0 +1,609 @@
+/// with only the sign changing between the two parties.
+struct Pnl(SignedAmount);
+
+impl Pnl {
+    /// Compute the PNL of the contract.
+    ///
+    /// Call `Self::new().long()` and `Self::new().short()` to access the PNL values for long and
+    /// short respectively.
+    fn new(
+        initial_price: u64,
+        closing_price: u64,
+        multiplier: Decimal,
+        n_contracts: u64,
+    ) -> Result<Self> {
+        let initial_price = Decimal::from(initial_price);
+        let closing_price = Decimal::from(closing_price);
+        let n_contracts = Decimal::from(n_contracts);
+
+        let pnl = (closing_price - initial_price) * multiplier * n_contracts;
+        let pnl = pnl.to_f64().context("Could not convert PNL to f64")?;
+        let pnl = SignedAmount::from_btc(pnl)
+            .context("Could not convert PNL to bitcoin::SignedAmount")?;
+
+        Ok(Self(pnl))
+    }
+
+    /// The profit and loss (PNL) from the perspective of the party going long.
+    fn long(&self) -> SignedAmount {
+        self.0
+    }
+
+    /// The profit and loss (PNL) from the perspective of the party going short.
+    fn short(&self) -> SignedAmount {
+        self.0 * -1
+    }
+}
+
+/// Closing prices at which parties get liquidated i.e. when they get a 0 payout.
+struct LiquidationThresholds {
+    inner: u64, // TODO: Find a better name!
+    initial_price: u64,
+}
+
+impl LiquidationThresholds {
+    /// Compute the liquidation thresholds for both parties.
+    ///
+    /// Call `Self::new().long()` and `Self::new().short()` to access the liquidation threshold
+    /// values for long and short respectively.
+    fn new(
+        initial_margin: Amount,
+        n_contracts: u64,
+        initial_price: u64,
+        multiplier: Decimal,
+    ) -> Result<Self> {
+        let initial_margin = Decimal::from_f64(initial_margin.as_btc())
+            .context("Could not create Decimal from initial margin")?;
+
+        let n_contracts = Decimal::from(n_contracts);
+
+        let inner = initial_margin / (multiplier * n_contracts);
+        let inner = inner
+            .to_u64()
+            .context("Could not convert inner threshold calculation to u64")?;
+
+        Ok(Self {
+            inner,
+            initial_price,
+        })
+    }
+
+    /// Compute the closing price under which the party going long gets liquidated.
+    fn long(&self) -> u64 {
+        self.initial_price.saturating_sub(self.inner)
+    }
+
+    /// Compute the closing price over which the party going short gets liquidated.
+    fn short(&self) -> u64 {
+        self.initial_price + self.inner
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::payouts::prop_compose::arb_fee_flow;
+    use crate::payouts::prop_compose::arb_leverage;
+    use crate::payouts::quanto;
+    use itertools::Itertools;
+    use proptest::prelude::*;
+    use rust_decimal_macros::dec;
+
+    #[test]
+    fn quanto_curve_snapshot() {
```

Comment:
Can we use our `backtesting` to verify this payout curve?

Reply:
Nothing stops us from doing so, the return type is the same if you call `model::Payouts::new_quanto`.

Do you want me to create a ticket?

Reply:
See https://github.com/itchysats/backtesting/issues/70.